bottlecap/src/bin/bottlecap/main.rs

```diff
@@ -330,154 +347,174 @@ async fn extension_loop_active(
         setup_telemetry_client(&r.extension_id, logs_agent_channel).await?;
 
     let mut flush_control = FlushControl::new(config.serverless_flush_strategy);
-    let mut shutdown = false;
 
     let mut flush_interval = flush_control.get_flush_interval();
     flush_interval.tick().await; // discard first tick, which is instantaneous
 
+    // first invoke we must call next
+    let next_lambda_response = next_event(client, &r.extension_id).await;
+
+    handle_next_invocation(next_lambda_response, invocation_processor.clone()).await;
     loop {
-        let evt = next_event(client, &r.extension_id).await;
-        match evt {
-            Ok(NextEventResponse::Invoke {
-                request_id,
-                deadline_ms,
-                invoked_function_arn,
-            }) => {
-                debug!(
-                    "Invoke event {}; deadline: {}, invoked_function_arn: {}",
-                    request_id, deadline_ms, invoked_function_arn
-                );
-                let mut p = invocation_processor.lock().await;
-                p.on_invoke_event(request_id);
-                drop(p);
-            }
-            Ok(NextEventResponse::Shutdown {
-                shutdown_reason,
-                deadline_ms,
-            }) => {
-                println!("Exiting: {shutdown_reason}, deadline: {deadline_ms}");
-                shutdown = true;
-            }
-            Err(err) => {
-                eprintln!("Error: {err:?}");
-                println!("Exiting");
-                return Err(err);
+        let shutdown;
+        // Don't await! Pin it and check in the tokio loop
+        // Tokio select! drops whatever task does not complete first
+        // but the state machine API in Lambda will give us an invalid error
+        // if we call /next twice instead of waiting
```

Comment:
Maybe this comment goes for the pin below?

Reply:
it did